pkg/ipamd/ipamd.go

```diff
@@ -884,26 +884,14 @@ func (c *IPAMContext) tryAllocateENI(ctx context.Context) error {
 		subnet = eniCfg.Subnet
 	}
 
-	eni, err := c.awsClient.AllocENI(c.useCustomNetworking, securityGroups, subnet)
+	resourcesToAllocate := c.GetENIResourcesToAllocate()
+	eni, err := c.awsClient.AllocENI(c.useCustomNetworking, securityGroups, subnet, resourcesToAllocate)
 	if err != nil {
 		log.Errorf("Failed to increase pool size due to not able to allocate ENI %v", err)
 		ipamdErrInc("increaseIPPoolAllocENI")
 		return err
 	}
 
-	resourcesToAllocate := c.GetENIResourcesToAllocate()
-	_, err = c.awsClient.AllocIPAddresses(eni, resourcesToAllocate)
-	if err != nil {
-		log.Warnf("Failed to allocate %d IP addresses on an ENI: %v", resourcesToAllocate, err)
-		// Continue to process the allocated IP addresses
-		ipamdErrInc("increaseIPPoolAllocIPAddressesFailed")
-		if containsInsufficientCIDRsOrSubnetIPs(err) {
```

Comment:
I think we still want to be able to log an error like this when the subnet is exhausted. Can you test what we will log after this change when the subnet does not have enough IPs? You can do this by creating small subnets and passing them to eksctl: https://eksctl.io/usage/creating-and-managing-clusters/